src/scripts/sync_synonym.py

```diff
@@ -0,0 +1,147 @@
+    # Iterate over mappings
+    mappings_df = pd.read_csv(mondo_mappings_path, sep='\t', comment='#')
+    # todo: refactor filtering by prefixes
+    #  check if this pattern has been used elsewhere; I think it has. and abstract into func for these 3 lines; 'filter prefixes'
+    #  - OR use this?: get_all_owned_terms()
+    #    - it seems long and messy compared to these 3 lines, but is more comprehensive. I could make get_all_owned_terms() a simple 3 line func and then make a separate func for all edge cases?
+    # - filter by source
+    owned_prefix_map: PREFIX_MAP = get_owned_prefix_map(onto_config_path)
+    mappings_df['object_prefix'] = mappings_df['object_id'].apply(lambda x: x.split(':')[0])
+    mappings_df = mappings_df[mappings_df['object_prefix'].isin(owned_prefix_map.keys())]
+    del mappings_df['object_prefix']
+
+    # Query synonyms
+    # - Mondo
+    mondo_df = pd.read_csv(mondo_scopes_and_xrefs_path, sep='\t').rename(columns={'?dbXref': 'source_id'})
+    mondo_df = mondo_df.rename(columns=lambda x: x.lstrip('?'))
+    mondo_df['object_prefix'] = mondo_df['source_id'].apply(lambda x: x.split(':')[0])
+    mondo_df = mondo_df[mondo_df['object_prefix'].isin(owned_prefix_map.keys())]
+    del mondo_df['object_prefix']
+    mondo_df['unprefixed_source_id'] = mondo_df['source_id'].apply(lambda x: x.split(':')[1])
+    mondo_df = mondo_df[mondo_df['unprefixed_source_id'].str.isnumeric()]
+    del mondo_df['unprefixed_source_id']
+
+    # - Source
+    source_db: SqlImplementation = get_adapter(ontology_db_path)
+    source_df: pd.DataFrame = _get_synonyms(mappings_df['object_id'].tolist(), source_db)
+    source_df = source_df.rename(columns={'curie': 'source_id'})[['synonym_scope', 'synonym', 'source_id']]
+
+    # Determine synchronization cases
+    confirmed_df = mondo_df.merge(source_df, on=['synonym_scope', 'synonym', 'source_id'], how='inner')
+    confirmed_df['case'] = 'confirmed'
+
+    updated_df = mondo_df.merge(source_df, on=['synonym', 'source_id'], how='inner').rename(
+        columns={'synonym_scope_x': 'synonym_scope_mondo', 'synonym_scope_y': 'synonym_scope'})
+    updated_df = updated_df[updated_df['synonym_scope_mondo'] != updated_df['synonym_scope']]
+    updated_df['case'] = 'updated'
+
+    # TODO: Filter by -updated
+    #  This needs to be filtered by some of the rows that appear in `-updated`. If the synonym exists in Mondo for a
+    #  given `source_id`, but has a different `synonym_scope`, it should not show up in this case.
+    added_df = source_df.merge(mondo_df, on=['synonym_scope', 'synonym', 'source_id'], how='left', indicator=True)
+    added_df = added_df[added_df['_merge'] == 'left_only'].drop('_merge', axis=1)  # also can be done via Mondo ID = nan
+    added_df['case'] = 'added'
+
+    # TODO: -deleted
+    deleted_df = mondo_df.merge(source_df, on=['synonym_scope', 'synonym', 'source_id'], how='left', indicator=True)
```

Comment:
## FYI: 'deleted' can also mean  'erroneously attributed'
By that, I mean that it could be the case that the source never ever had that combination of scope+synonym, and when we added it to Mondo, it was done so erroneously. Though I imagine that will not be the case very often, if at all. In any case I'll add a comment to the code about that.

Reply:
FYI: 'deleted' + 'added' can also mean 'mispelling or alternative case had been previously added'
For example, I see that for OMIM:227646, Mondo had **Fad2**, but OMIM had **fad2**.

I don't know if this is because (a) OMIM changed to lowercase, or (b) Mondo had previously added this synonym, but accidentally did it in sentence case.

Reply:
We should not analyse old attributions, they are too chaotic. Dont even query for them. Just query for synonym types (abbreviation etc) and properties (exact/related/ etc) - that is all you need.

Reply:
Re capitalisation: 

@twhetzel and I discussed that for now, it is ok to add both capitalisations side by side. However, of course, when determining whether the synonym scope (broad/narrow/exact) changed, we should take capitalisation out of the equation because it would be silly to have Fad2 be exact, and fad2 to be related.

Reply:
- [ ] @joeflack4 Handle any issues in regards to capitalization

RE: attributions
@matentzn I think what you define as attribution is different than what I'm thinking. What I mean by "erroneous attribution" is simply where we say that we got x synonym from OMIM:123, but we were wrong and OMIM:123 never had that synonym.
I'm assuming that there is some sort of "attribution" annotation property / axiom situation that I don't know about which is what you are thinking of.

RE: capitalization
What you say makes sense. I'll do that, but I've marked it as a lower priority.
I have a feeling though that there will be more to think about in regards to capitalization. For example, I feel like there could be some edge cases where a difference in case might indicate a different synonym entirely, rather than an alternative casing, e.g. in the case of acronyms.

Reply:
RE: capitalization
I'm just starting to get to this. 

- [ ] @matentzn or @twhetzel Confirm that my plan below is good.

Nico said:
> @twhetzel and I discussed that for now, it is ok to add both capitalisations side by side. However, of course, when determining whether the synonym scope (broad/narrow/exact) changed, we should take capitalisation out of the equation because it would be silly to have Fad2 be exact, and fad2 to be related.

I don't think that this is an issue. Here's are several illustrative cases representing current behavior:
- If only synonym scope differs, this is considered `updated`.
- If only spelling / capitalization differs, the spelling in Mondo will be `deleted` and the source spelling will be `added`.
- If spelling / capitalization _and_ scope differ, the scope+spelling in Mondo will be `deleted` and the source scope+spelling will be `added`.

Reply:
We are reviewing this.

Reply:
> @matentzn I think what you define as attribution is different than what I'm thinking. What I mean by "erroneous attribution" is simply where we say that we got x synonym from OMIM:123, but we were wrong and OMIM:123 never had that synonym.

No, this is exactly what I meant: "we got x synonym from OMIM:123" - this information should not be important to your task. Moving this discussion to slack now.